MarvelApiClient/src/main/java/com/karumi/marvelapiclient/model/ComicsQuery.java

```diff
@@ -0,0 +1,574 @@
+    this.ean = ean;
+    this.issn = issn;
+    this.hasDigitalIssue = hasDigitalIssue;
+    this.modifiedSince = modifiedSince;
+    this.creators = creators;
+    this.characters = characters;
+    this.series = series;
+    this.events = events;
+    this.stories = stories;
+    this.sharedAppearances = sharedAppearances;
+    this.collaborators = collaborators;
+    this.orderBy = orderBy;
+    this.limit = limit;
+    this.offset = offset;
+  }
+
+  public Map<String, Object> toMap() {
+    Map<String, Object> returnValues = new HashMap<>();
+
+    if (format != null) {
+      returnValues.put(QUERY_FORMAT, format);
+    }
+
+    if (formatType != null) {
+      returnValues.put(QUERY_FORMAT_TYPE, formatType);
+    }
+
+    if (noVariants) {
+      returnValues.put(QUERY_NO_VARIANTS, noVariants);
+    }
+
+    if (dateDescriptor != null) {
+      returnValues.put(QUERY_DATE_DESCRIPTOR, dateDescriptor);
+    }
+
+    if (dateRange != null) {
+      returnValues.put(QUERY_DATE_RANGE, dateRange);
+    }
+    if (title != null) {
+      returnValues.put(QUERY_TITLE, title);
+    }
+    if (titleStartsWith != null) {
+      returnValues.put(QUERY_TITLE_STARTS_WITH, titleStartsWith);
+    }
+
+    if (startYear > 0) {
+      returnValues.put(QUERY_START_YEAR, startYear);
+    }
+
+    if (issueNumber > 0) {
+      returnValues.put(QUERY_ISSUE_NUMBER, issueNumber);
+    }
+
+    if (diamondCode != null) {
+      returnValues.put(QUERY_DIAMOND_CODE, diamondCode);
+    }
+
+    if (digitalId > 0) {
+      returnValues.put(QUERY_DIGITAL_ID, digitalId);
+    }
+
+    if (upc != null) {
+      returnValues.put(QUERY_UPC, upc);
+    }
+
+    if (isbn != null) {
+      returnValues.put(QUERY_ISBN, isbn);
+    }
+
+    if (ean != null) {
+      returnValues.put(QUERY_EAN, ean);
+    }
+
+    if (issn != null) {
+      returnValues.put(QUERY_ISSN, issn);
+    }
+
+    if (hasDigitalIssue) {
+      returnValues.put(QUERY_HAS_DIGITAL_ISSUE, hasDigitalIssue);
+    }
+
+    if (modifiedSince != null) {
+      returnValues.put(QUERY_MODIFIED_SINCE, modifiedSince);
+    }
+
+    if (creators != null) {
+      returnValues.put(QUERY_CREATORS, creators);
+    }
+
+    if (characters != null) {
+      returnValues.put(QUERY_CHARACTERS, characters);
+    }
+
+    if (series != null) {
+      returnValues.put(QUERY_SERIES, series);
+    }
+
+    if (events != null) {
+      returnValues.put(QUERY_EVENTS, events);
+    }
+
+    if (stories != null) {
+      returnValues.put(QUERY_STORIES, stories);
+    }
+
+    if (sharedAppearances != null) {
+      returnValues.put(QUERY_SHARED_APPEARANCES, sharedAppearances);
+    }
+
+    if (collaborators != null) {
+      returnValues.put(QUERY_COLLABORATORS, collaborators);
+    }
+
+    if (orderBy != null) {
+      returnValues.put(QUERY_ORDER_BY, orderBy);
+    }
+
+    if (limit > 0) {
+      returnValues.put(QUERY_LIMIT, limit);
+    }
+
+    if (offset > 0) {
+      returnValues.put(QUERY_OFFSET, offset);
+    }
```

Comment:
What's the problem adding null values to the map? Retrofit should not send null values. In the case of the limit or the offset you can declare it as ``Integer`` and avoid all these non null checks.